### PR TITLE
Update Example Project

### PR DESCRIPTION
**Pull request by joshuakg**
Updating the example project dependencies from:
```
"react": "17.0.2",
"react-native": "0.66.4",
```
to
```
"react":  "18.2.0",
"react-native": "0.73.6"
```

### CODE REVIEW

**Review by TannerDrayton — APPROVED** [review]

File: example/ios/Podfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ if linkage != nil
   use_frameworks! :linkage => linkage.to_sym
 end
 
-target 'ExampleApp' do
+target 'exampleApp' do
```
Comment (calebmackdavenport) [inline example/ios/Podfile:28]: We could remove flipper. It was fixed specifically for 73. That fix was not backported to 71 or 72. And it is removed entirely in 74.

File: example/package.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-  "name": "ExampleApp",
+  "name": "exampleApp",
   "version": "0.0.1",
   "private": true,
   "scripts": {
     "android": "react-native run-android",
     "ios": "react-native run-ios",
     "lint": "eslint .",
-    "pod": "(rm -rf ios/Pods && cd ios/ && pod install --repo-update)",
+    "pod": "(rm -rf ios/Pods && cd ios/ && pod install --repo-update && pod update hermes-engine --no-repo-update)",
```
Comment (calebmackdavenport) [inline example/package.json:9]: Obviously less than ideal to have to do this, but it does appear to remedy with the only byproduct being downloading the tarball twice

**Review by calebmackdavenport — APPROVED** [review]

**Comment by calebmackdavenport** [comment]
👍🏼

**Review by jkhusanov — APPROVED** [review]
🙌

**Pull request merged**